components/script/dom/htmlinputelement.rs

```diff
@@ -81,7 +86,7 @@ pub struct HTMLInputElement {
     // https://html.spec.whatwg.org/multipage/#concept-input-value-dirty-flag
     value_dirty: Cell<bool>,
 
-    // TODO: selected files for file input
+    filelist: DOMRefCell<Option<JS<FileList>>>,
```

Comment:
Not sure if this needs to be Option in the first place, really. Helps avoid unnecessary heap allocation, but there may be more holistic solutions for this.